### ISSUE TITLE
[auto_submit] Include commit description when auto-merging PRs

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#103540]
## Problem statement
Today, PRs that are merged by the Flutter auto-merge bot (when tagged with the `waiting for tree to go green` label), have the PR description as their title, but drop the commit description. It would be a huge improvement if commit description were included.

## Rationale for change
The description is a key part of a commit that often includes critical information that isn't part of the code or comments and that can be incredibly important to future code archaeologists fixing bugs or understanding a given piece of code. These include:
* Rationale for the change. What prompted it and why? What was the context? [Example](https://github.com/flutter/engine/pull/33127)
* Relevant information for understanding the change. [Example](https://github.com/flutter/engine/pull/30805)
* Links to design documents. [Example](https://github.com/flutter/engine/pull/19024)
* In the case of re-lands, links to the revert and an explanation of what went wrong the first time the code was landed, and how it was fixed. [Example](https://github.com/flutter/engine/pull/33238)
* Links to relevant issues that prompted this. (All PRs are meant to include this)

In a lot of cases, good commit descriptions can save hours of time; when narrowing a piece of code down to a couple lines of code, understanding why that code was written can be critical to understanding whether it's still relevant and what to consider when changing it. 

## Potential implementations
In terms of implementation I'd suggest one of the following, in order of (personal) preference:
1. The commit comment of the first commit in the PR. For the frequent degenerate case where only one commit exists, that's the one to use.
2. The PR description, ideally minus the checkbox boilerplate.
3. The concatenation of all commit comments, separated by a blank line. This is GitHub's default implementation for the manual merge box, but can lead to a nice description followed by a half dozen "Review fixes" type comments.

If nothing else, I think implementing case 1 or 2 would provide a huge win. 

/cc @CaseyHillers @Hixie

**Comment by CaseyHillers** [comment]
\cc @keyonghan @godofredoc

**Comment by stuartmorgan** [comment]
Duplicate of https://github.com/flutter/flutter/issues/49388 ?

**Comment by cbracken** [comment]
Indeed. And apparently I commented on this at the time and forgot about it :/ Was fresh in my mind since I'd spent a couple hours trying to figure out why a few lines of code had been added a long time ago but noticing a lot of the commits that looked relevant had no descriptions, resulting in a lot of manual URL construction to check if the PR descriptions might have had the info I needed. Closing in favour of that.
